mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/uwazi/entry/UwaziSelectEntitiesFragment.kt

```diff
@@ -0,0 +1,148 @@
+package rs.readahead.washington.mobile.views.fragment.uwazi.entry
+
+import android.graphics.Color
+import android.os.Bundle
+import android.view.View
+import android.widget.EditText
+import androidx.appcompat.widget.SearchView
+import androidx.fragment.app.setFragmentResult
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import rs.readahead.washington.mobile.R
+import rs.readahead.washington.mobile.data.uwazi.UwaziConstants
+import rs.readahead.washington.mobile.data.uwazi.UwaziConstants.UWAZI_RELATION_SHIP_REQUEST_KEY
+import rs.readahead.washington.mobile.databinding.UwaziSelectEntitiesFragmentBinding
+import rs.readahead.washington.mobile.presentation.uwazi.UwaziRelationShipEntity
+import rs.readahead.washington.mobile.views.base_ui.BaseBindingDialogFragment
+import rs.readahead.washington.mobile.views.fragment.uwazi.widgets.searchable_multi_select.SearchableAdapter
+import rs.readahead.washington.mobile.views.fragment.uwazi.widgets.searchable_multi_select.SearchableItem
+
+
+class UwaziSelectEntitiesFragment :
+    BaseBindingDialogFragment<UwaziSelectEntitiesFragmentBinding>(UwaziSelectEntitiesFragmentBinding::inflate) {
+    private var items: MutableList<SearchableItem> = ArrayList()
+    private var resultList: MutableList<UwaziRelationShipEntity> = ArrayList()
+    private val uwaziParser: UwaziParser by lazy { UwaziParser(context) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(
+            STYLE_NORMAL,
+            R.style.FullScreenDialogStyle
+        );
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initializeData()
+        initViews()
+    }
+
+    private fun initializeData() {
+        if (arguments?.getString(UWAZI_TEMPLATE) != null) {
+            arguments?.getString(UWAZI_TEMPLATE)?.let {
+                uwaziParser.parseTemplateForRelationShipEntities(it)
+            }
+        }
+
+        if (arguments?.getString(UWAZI_ENTRY_PROMPT_ID) != null) {
+            val id = arguments?.getString(UWAZI_ENTRY_PROMPT_ID)
+            val property =
+                uwaziParser.getTemplate()?.entityRow?.properties?.find { property -> property._id == id }
+            var i = 0
+            property?.entities?.forEach { value ->
+                items.add(SearchableItem(value.label, value.id, i.toString()))
+                i++
+            }
+        }
+
+        if (arguments?.getString(UWAZI_SELECTED_ENTITIES) != "[]") {
```

Comment:
let use scope function `takeif`  `.takeIf { it != "[]" }`

Reply:
and later we can access it and check if it's nullable -> ?.let { selectedEntitiesJson ->